```
django-hijack> Executing setuptoolsBuildPhase
django-hijack> total 67K
django-hijack> drwxr-xr-x   2 nixbld nixbld  24 May 16 15:47 .
django-hijack> drwxr-xr-x 326 nixbld nixbld 327 May 16 15:47 ..
django-hijack> lrwxrwxrwx   1 nixbld nixbld  18 May 16 15:47 acorn -> ../acorn/bin/acorn
django-hijack> lrwxrwxrwx   1 nixbld nixbld  32 May 16 15:47 autoprefixer -> ../autoprefixer/bin/autoprefixer
django-hijack> lrwxrwxrwx   1 nixbld nixbld  22 May 16 15:47 browserslist -> ../browserslist/cli.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  32 May 16 15:47 browserslist-lint -> ../update-browserslist-db/cli.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  20 May 16 15:47 cssesc -> ../cssesc/bin/cssesc
django-hijack> lrwxrwxrwx   1 nixbld nixbld  22 May 16 15:47 envinfo -> ../envinfo/dist/cli.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  23 May 16 15:47 eslint -> ../eslint/bin/eslint.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  31 May 16 15:47 import-local-fixture -> ../import-local/fixtures/cli.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  25 May 16 15:47 js-yaml -> ../js-yaml/bin/js-yaml.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  19 May 16 15:47 json5 -> ../json5/lib/cli.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  22 May 16 15:47 loose-envify -> ../loose-envify/cli.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  24 May 16 15:47 nanoid -> ../nanoid/bin/nanoid.cjs
django-hijack> lrwxrwxrwx   1 nixbld nixbld  22 May 16 15:47 resolve -> ../resolve/bin/resolve
django-hijack> lrwxrwxrwx   1 nixbld nixbld  16 May 16 15:47 rimraf -> ../rimraf/bin.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  15 May 16 15:47 sass -> ../sass/sass.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  23 May 16 15:47 semver -> ../semver/bin/semver.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  22 May 16 15:47 standard -> ../standard/bin/cmd.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  29 May 16 15:47 stylelint -> ../stylelint/bin/stylelint.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  20 May 16 15:47 terser -> ../terser/bin/terser
django-hijack> lrwxrwxrwx   1 nixbld nixbld  25 May 16 15:47 webpack -> ../webpack/bin/webpack.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  25 May 16 15:47 webpack-cli -> ../webpack-cli/bin/cli.js
django-hijack> lrwxrwxrwx   1 nixbld nixbld  18 May 16 15:47 which -> ../which/bin/which
django-hijack> /nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/config/setupcfg.py:520: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
django-hijack>   warnings.warn(msg, warning_class)
django-hijack> /nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/__init__.py:85: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
django-hijack>   dist.fetch_build_eggs(dist.setup_requires)
django-hijack> /nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/config/setupcfg.py:520: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
django-hijack>   warnings.warn(msg, warning_class)
django-hijack> running bdist_wheel
django-hijack> running build
django-hijack> running build_py
django-hijack> creating build
django-hijack> creating build/lib
django-hijack> creating build/lib/hijack
django-hijack> copying hijack/conf.py -> build/lib/hijack
django-hijack> copying hijack/apps.py -> build/lib/hijack
django-hijack> copying hijack/__init__.py -> build/lib/hijack
django-hijack> copying hijack/middleware.py -> build/lib/hijack
django-hijack> copying hijack/signals.py -> build/lib/hijack
django-hijack> copying hijack/urls.py -> build/lib/hijack
django-hijack> copying hijack/permissions.py -> build/lib/hijack
django-hijack> copying hijack/views.py -> build/lib/hijack
django-hijack> running egg_info
django-hijack> creating django_hijack.egg-info
django-hijack> writing django_hijack.egg-info/PKG-INFO
django-hijack> writing dependency_links to django_hijack.egg-info/dependency_links.txt
django-hijack> writing requirements to django_hijack.egg-info/requires.txt
django-hijack> writing top-level names to django_hijack.egg-info/top_level.txt
django-hijack> writing manifest file 'django_hijack.egg-info/SOURCES.txt'
django-hijack> reading manifest file 'django_hijack.egg-info/SOURCES.txt'
django-hijack> reading manifest template 'MANIFEST.in'
django-hijack> no previously-included directories found matching '.github'
django-hijack> warning: no previously-included files found matching '.gitignore'
django-hijack> warning: no previously-included files found matching 'requirements.txt'
django-hijack> warning: no files found matching 'hijack/locale/*/LC_MESSAGES/django.mo'
django-hijack> warning: no files found matching 'hijack/static/hijack/hijack.min.*'
django-hijack> adding license file 'LICENSE'
django-hijack> writing manifest file 'django_hijack.egg-info/SOURCES.txt'
django-hijack> running compile_translations
django-hijack> running command: msgfmt -c -o hijack/locale/sk/LC_MESSAGES/django.mo hijack/locale/sk/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/da/LC_MESSAGES/django.mo hijack/locale/da/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/ja/LC_MESSAGES/django.mo hijack/locale/ja/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/de/LC_MESSAGES/django.mo hijack/locale/de/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/ru/LC_MESSAGES/django.mo hijack/locale/ru/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/cs/LC_MESSAGES/django.mo hijack/locale/cs/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/fr/LC_MESSAGES/django.mo hijack/locale/fr/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/pt_BR/LC_MESSAGES/django.mo hijack/locale/pt_BR/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/nl/LC_MESSAGES/django.mo hijack/locale/nl/LC_MESSAGES/django.po
django-hijack> running command: msgfmt -c -o hijack/locale/es/LC_MESSAGES/django.mo hijack/locale/es/LC_MESSAGES/django.po
django-hijack> running compile_scss
django-hijack> running command: npm ci
django-hijack> 
django-hijack> added 449 packages, and audited 450 packages in 5s
django-hijack> 
django-hijack> 99 packages are looking for funding
django-hijack>   run `npm fund` for details
django-hijack> 
django-hijack> found 0 vulnerabilities
django-hijack> running command: npm run build
django-hijack> 
django-hijack> > django-hijack@1.0.0 build
django-hijack> > webpack --mode=production
django-hijack> 
django-hijack> sh: line 1: /build/source/node_modules/.bin/webpack: cannot execute: required file not found
django-hijack> Traceback (most recent call last):
django-hijack>   File "/build/source/nix_run_setup", line 8, in <module>
django-hijack>     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
django-hijack>   File "setup.py", line 74, in <module>
django-hijack>     setup(
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/__init__.py", line 108, in setup
django-hijack>     return distutils.core.setup(**attrs)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
django-hijack>     return run_commands(dist)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
django-hijack>     dist.run_commands()
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
django-hijack>     self.run_command(cmd)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/dist.py", line 1221, in run_command
django-hijack>     super().run_command(command)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
django-hijack>     cmd_obj.run()
django-hijack>   File "/nix/store/ylz3xaryjhskfjbwc2p069zv9q2b3jj8-python3.10-wheel-0.38.4/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 325, in run
django-hijack>     self.run_command("build")
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
django-hijack>     self.distribution.run_command(command)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/dist.py", line 1221, in run_command
django-hijack>     super().run_command(command)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
django-hijack>     cmd_obj.run()
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 131, in run
django-hijack>     self.run_command(cmd_name)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
django-hijack>     self.distribution.run_command(command)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/dist.py", line 1221, in run_command
django-hijack>     super().run_command(command)
django-hijack>   File "/nix/store/a9nrh0jkfxcf4v1arm0v8cg491wpi6y9-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
django-hijack>     cmd_obj.run()
django-hijack>   File "setup.py", line 53, in run
django-hijack>     subprocess.check_call(
django-hijack>   File "/nix/store/95cxzy2hpizr23343b8bskl4yacf4b3l-python3-3.10.11/lib/python3.10/subprocess.py", line 369, in check_call
django-hijack>     raise CalledProcessError(retcode, cmd)
django-hijack> subprocess.CalledProcessError: Command '['npm', 'run', 'build']' returned non-zero exit status 127.

--- a/pkgs/development/python-modules/django-hijack-admin/default.nix
+++ b/pkgs/development/python-modules/django-hijack-admin/default.nix
@@ -1,5 +1,5 @@
 { lib, buildPythonPackage, fetchFromGitHub, python,
-  django_hijack, django_nose }:
+  django-hijack, django_nose }:
 buildPythonPackage rec {
   pname = "django-hijack-admin";
   version = "2.1.10";
@@ -13,7 +13,7 @@ buildPythonPackage rec {
   };
 
   nativeCheckInputs = [ django_nose ];
-  propagatedBuildInputs = [ django_hijack ];
+  propagatedBuildInputs = [ django-hijack ];
 
   checkPhase = ''
     runHook preCheck

--- a/pkgs/development/python-modules/django-hijack/default.nix
+++ b/pkgs/development/python-modules/django-hijack/default.nix
@@ -1,27 +1,56 @@
 { lib
-, fetchPypi
+, fetchFromGitHub
+, fetchNpmDeps
 , buildPythonPackage
+
+# build-system
+, gettext
+, nodejs
+, npmHooks
+, setuptools-scm
+
+# dependencies
 , django
 , django_compat
+
+# tests
 , pytest-django
 , pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "django-hijack";
-  version = "3.2.6";
+  version = "3.3.0";
+  format = "setuptools";
 
-  # the wheel comes with pre-built assets, allowing us to avoid fighting
-  # with npm/webpack/gettext to build them ourselves.
-  format = "wheel";
-
-  src = fetchPypi {
-    inherit version format;
-    pname = "django_hijack";
-    dist = "py3";
-    python = "py3";
-    hash = "sha256-xFPZ03II1814+bZ5gx7GD/AxYMiLuH6awfSeXEraOHQ=";
+  src = fetchFromGitHub {
+    owner = "django-hijack";
+    repo = "django-hijack";
+    rev = "refs/tags/${version}";
+    hash = "sha256-ytQ4xxkBAC3amQbenD8RO5asrbfNAjOspWUY3c2hkig=";
   };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace 'cmd = ["npm", "ci"]' 'cmd = ["true"]' \
+      --replace 'f"{self.build_lib}/{name}.mo"' 'f"{name}.mo"'
+
+    sed -i "/addopts/d" setup.cfg
+  '';
+
+  npmDeps = fetchNpmDeps {
+    inherit src;
+    hash = "sha256-FLfMCn2jsLlTTsC+LRMX0dmVCCbNAr2pQUsSQRKgo6E=";
+  };
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [
+    gettext
+    nodejs
+    npmHooks.npmConfigHook
+    setuptools-scm
+  ];
 
   propagatedBuildInputs = [
     django
@@ -33,13 +62,11 @@ buildPythonPackage rec {
     pytest-django
   ];
 
-  preCheck = ''
-    export DJANGO_SETTINGS_MODULE='hijack.tests.test_app.settings'
-  '';
+  env.DJANGO_SETTINGS_MODULE = "hijack.tests.test_app.settings";
 
   pytestFlagsArray = [
-    "--pyargs"
-    "hijack"
+    "--pyargs" "hijack"
+    "-W" "ignore::DeprecationWarning"
   ];
 
   meta = with lib; {

--- a/pkgs/top-level/python-aliases.nix
+++ b/pkgs/top-level/python-aliases.nix
@@ -86,6 +86,7 @@ mapAliases ({
   django_extensions = django-extensions; # added 2022-01-09
   django_guardian = django-guardian; # added 2022-05-19
   django_hijack = django-hijack; # added 2023-05-16
+  django_hijack_admin = django-hijack-admin; # added 2023-05-16
   django_modelcluster = django-modelcluster; # added 2022-04-02
   django_reversion = django-reversion; # added 2022-06-18
   django_polymorphic = django-polymorphic; # added 2022-05-24

--- a/pkgs/top-level/python-aliases.nix
+++ b/pkgs/top-level/python-aliases.nix
@@ -85,6 +85,7 @@ mapAliases ({
   django_environ = django-environ; # added 2021-12-25
   django_extensions = django-extensions; # added 2022-01-09
   django_guardian = django-guardian; # added 2022-05-19
+  django_hijack = django-hijack; # added 2023-05-16
   django_modelcluster = django-modelcluster; # added 2022-04-02
   django_reversion = django-reversion; # added 2022-06-18
   django_polymorphic = django-polymorphic; # added 2022-05-24

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -2691,7 +2691,7 @@ self: super: with self; {
 
   django-hierarkey = callPackage ../development/python-modules/django-hierarkey { };
 
-  django_hijack_admin = callPackage ../development/python-modules/django-hijack-admin { };
+  django-hijack-admin = callPackage ../development/python-modules/django-hijack-admin { };
 
   django-hijack = callPackage ../development/python-modules/django-hijack { };
 

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -2693,8 +2693,7 @@ self: super: with self; {
 
   django_hijack_admin = callPackage ../development/python-modules/django-hijack-admin { };
 
-  django_hijack = callPackage ../development/python-modules/django-hijack { };
-  # This package may need an older version of Django. Override the package set and set e.g. `django = super.django_1_9`. See the Nixpkgs manual for examples on how to override the package set.
+  django-hijack = callPackage ../development/python-modules/django-hijack { };
 
   django-i18nfield = callPackage ../development/python-modules/django-i18nfield { };
 

